src/app/Dashboard/AddCard.tsx

```diff
@@ -157,3 +233,124 @@ export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardPro
     </>
   );
 };
+
+interface PropsConfigFormProps {
+  cardTitle: string;
+  controls: PropControl[];
+  initialState: any;
+  onChange: ({}) => void;
+}
+
+const PropsConfigForm = (props: PropsConfigFormProps) => {
+  const [propsConfig, setPropsConfig] = React.useState(props.initialState);
+
+  const handleChange = React.useCallback(
+    (k) => (e) => {
+      setPropsConfig((old) => {
+        const copy = { ...old };
+        copy[k] = e;
+        props.onChange(copy);
+        return copy;
+      });
+    },
+    [props.onChange, setPropsConfig]
+  );
+
+  const handleNumeric = React.useCallback(
+    (k) => (e) => {
+      const value = (e.target as HTMLInputElement).value;
+      setPropsConfig((old) => {
+        const copy = { ...old };
+        copy[k] = value;
+        props.onChange(copy);
+        return copy;
+      });
+    },
+    [props.onChange, setPropsConfig]
+  );
+
+  const handleNumericStep = React.useCallback(
+    (k, v) => (e) => {
+      setPropsConfig((old) => {
+        const copy = { ...old };
+        copy[k] = old[k] + v;
+        props.onChange(copy);
+        return copy;
+      });
+    },
+    [props.onChange, setPropsConfig]
+  );
+
+  const createControl = React.useCallback(
+    (ctrl: PropControl): JSX.Element => {
+      let input: JSX.Element;
+      switch (ctrl.kind) {
+        case 'boolean':
+          input = (
+            <Switch label={ctrl.name} isReversed isChecked={propsConfig[ctrl.key]} onChange={handleChange(ctrl.key)} />
```

Comment:
I think the `label/labelOff` on `Switch` are messages for when the switch is on/off? The `ctrl.name` should be part of the FormGroup label?

https://www.patternfly.org/v4/components/switch#switch

Reply:
Also, no reverse too? Just thinking we could keep it similar to Openshift console.


![Screenshot from 2022-12-12 20-08-05](https://user-images.githubusercontent.com/68053619/207201285-bea09c65-fa97-4bad-80c0-885119a36e9e.png)

Reply:
![Screenshot from 2022-12-12 20-27-52](https://user-images.githubusercontent.com/68053619/207203630-45b25dab-33c7-44ce-aa8c-e5cdb1ecf570.png)

Reply:
Oh right, thanks. These things were all pretty broken before when I had the nested formgroup setup, but now it works properly.